src/sagas/product.saga.js

```diff
@@ -0,0 +1,32 @@
+import { put, call, takeEvery } from 'redux-saga/effects';
+import {  productApiData } from "../actions/index";
+import { REQUEST_API_DATA } from '../constants/action-types';
+
+const fetchApiData = async ()=>{
```

Comment:
There is no need for a try/catch here, this function is not a generator function